e2e/test/iothub/service/ConfigurationsClientE2ETests.cs

```diff
@@ -86,5 +88,162 @@ public async Task ConfigurationOperations_Work()
+            }
+        }
+
+        [LoggedTestMethod, Timeout(TestTimeoutMilliseconds)]
+        public async Task ConfigurationsClient_DeleteETag_Works()
+        {
+            bool configCreated = false;
+            using var serviceClient = new IotHubServiceClient(TestConfiguration.IoTHub.ConnectionString);
+            string configurationId = (_idPrefix + Guid.NewGuid()).ToLower(); // Configuration Id characters must be all lower-case.
+            try
+            {
+                var configuration = new Configuration(configurationId)
+                {
+                    Priority = 2,
+                    Labels = { { "labelName", "labelValue" } },
+                    TargetCondition = "deviceId='fakeDevice'",
+                    Content = new ConfigurationContent
+                    {
+                        DeviceContent = { { "properties.desired.x", 4L } },
+                    },
+                    Metrics = new ConfigurationMetrics
+                    {
+                        Queries = { { "successfullyConfigured", "select deviceId from devices where properties.reported.x = 4" } }
+                    },
+                };
+
+                configuration = await serviceClient.Configurations.CreateAsync(configuration).ConfigureAwait(false);
+                configCreated = true;
+                ETag oldEtag = configuration.ETag;
+
+                configuration.Priority = 3;
+
+                configuration = await serviceClient.Configurations.SetAsync(configuration).ConfigureAwait(false);
+
+                configuration.ETag = oldEtag;
+
+                try
+                {
+                    // set the 'onlyIfUnchanged' flag to true to check that, with an out of date ETag, the request throws a PreconditionFailedException.
+                    await serviceClient.Configurations.DeleteAsync(configuration, true).ConfigureAwait(false);
+                    throw new AssertFailedException("Expected test to throw a precondition failed exception since it updated a configuration with an out of date ETag");
+                }
+                catch (DeviceMessageLockLostException)
+                {
+                    // expected thrown exception, continue the test without throwing anything
+                }
+
+                try
+                {
+                    // set the 'onlyIfUnchanged' flag to false to check that, even with an out of date ETag, the request performs without exception.
+                    await serviceClient.Configurations.DeleteAsync(configuration, false).ConfigureAwait(false);
+                }
+                catch (DeviceMessageLockLostException)
+                {
+                    throw new AssertFailedException("Did not expect test to throw a precondition failed exception since 'onlyIfUnchanged' was set to false");
+                }
+            }
+            finally
+            {
+                try
+                {
+                    if (configCreated)
```

Comment:
No need to check this. Just attempt the delete operation and let it throw a not found exception that the test ignores